File: Chuong 11 - Ke Thua/Ung Dung Quan Ly Thu Vien C++ - C# - Java/Source Code/C++/Chuong Trinh Quan Ly Doc Gia/QuanLyDocGia/DocGiaTreEm.cpp
#include "DocGiaTreEm.h"

istream& operator >>(istream &is, DocGiaTreEm &x)
{
	// Gọi lại operator nhập của cha
	DocGia *cha = static_cast<DocGia *>(&x);
	is >> *cha; 

	fflush(stdin);
	cout << "\nNhap vao ho ten nguoi dai dien: ";
	getline(is, x.HoTenNguoiDaiDien);

	return is;
}

ostream& operator <<(ostream &os, DocGiaTreEm x)
{
	// Gọi lại operator xuất của cha
	DocGia cha = static_cast<DocGia>(x);
	os << cha;

	os << "\nHo ten nguoi dai dien: " << x.HoTenNguoiDaiDien;
	os << "\nTien lam the: " << (size_t)x.TinhTienLamThe();

	return os;
}

float DocGiaTreEm::TinhTienLamThe()
{
	return 20000;
}

DocGiaTreEm::DocGiaTreEm(void)
{
}


DocGiaTreEm::~DocGiaTreEm(void)
{
}

File: Chuong 11 - Ke Thua/Ung Dung Quan Ly Thu Vien C++ - C# - Java/Source Code/Java/Java_QuanLyDocGia/src/java_quanlydocgia/DocGiaTreEm.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package java_quanlydocgia;

import java.io.Console;
import java.util.Scanner;

/**
 *
 * @author NguyenVietNamSon
 */
public class DocGiaTreEm extends DocGia{
    private String HoTenNguoiDaiDien;

        DocGia x = new DocGia();
        public void Nhap()
        {
            
            x.Nhap();

            Scanner scan = new Scanner(System.in);
            System.out.print("\nNhap vao ho ten nguoi dai dien: ");
            HoTenNguoiDaiDien = scan.nextLine();
        }

        public void Xuat()
        {
            x.Xuat();

            System.out.print("\nHo ten nguoi dai dien: " + String.valueOf(HoTenNguoiDaiDien));
            System.out.print("\nTien lam the: " + String.valueOf(TinhTienLamThe()));
            
        }

        public float TinhTienLamThe()
        {
            return 20000;
        }
}
